client/lib/signup/step-actions/index.js

```diff
@@ -203,8 +203,9 @@ export function createSiteWithCart( callback, dependencies, stepData, reduxStore
 
 	const shouldSkipDomainStep = ! siteUrl && isDomainStepSkippable( flowToCheck );
 	const shouldHideFreePlan = get( signupDependencies, 'shouldHideFreePlan', false );
+	const shouldHideDomainStep = ! siteUrl && 'onboarding-plan-first' === flowToCheck;
```

Comment:
What's the difference between `shouldHideDomainStep` and `shouldSkipDomainStep`? Could we merge this logic into `isDomainStepSkippable()`?

Reply:
Yea I can see how the variable names are similar sounding here. There is a slight difference between the two - `shouldSkipDomainStep` is in the case where we add a Skip button to the domain step. So the step is visible to the user but they can skip it if they want(this option is disabled at the moment). `shouldHideDomainStep` is the case where we don't even show the domain step to the user. We could of course combine them with 
```
! siteUrl && ( isDomainStepSkippable( flowToCheck ) || 'onboarding-plan-first' === flowToCheck ). 
```

Reply:
Cool. No objection on keeping it as it is :)